src/components/Chat/ChatSession.js

```diff
@@ -461,13 +497,19 @@ class ChatSession {
     }
     return this.client
       .getTranscript(args)
-      .then((response) => {
+      .then(async response => {
         var incomingDataList = response.data.Transcript;
         this.nextToken = response.data.NextToken;
         const transcriptItems = incomingDataList.map((data) => {
           var transcriptItem = modelUtils.createItemFromIncoming(data, this.thisParticipant);
           return transcriptItem;
         });
+        // call describeview to get the view if the newest message is a view message
```

Comment:
Why don't we have a copy of this logic in ChatJS, so customer does not have to mutate the front-end transcript object ?
currently, we pop, modify object and push . 
This logic can be moved to ChatJS, so `onInteractiveMessage` callback, we make describeView API call and update response with viewData ?

Reply:
This logic is only used to load a transcript on the front end side. But describeAndProcessView can potentially be moved to chatJS